syntax = "proto3";

package dbt_core_events;

import "google/protobuf/timestamp.proto";


message A001 {
    string code = 1;
    string msg = 2;
    string level = 3;
    string invocation_id = 4;
    int32 pid = 5;
    string thread_name = 6;
    google.protobuf.Timestamp ts = 7;
    string version = 10;
}

message A002 {
    string code = 1;
    string msg = 2;
    string level = 3;
    string invocation_id = 4;
    int32 pid = 5;
    string thread_name = 6;
    google.protobuf.Timestamp ts = 7;
    map<string, string> args = 10;  // Dict[str, str]
}

message E009 {
    string code = 1;
    string msg = 2;
    string level = 3;
    string invocation_id = 4;
    int32 pid = 5;
    string thread_name = 6;
    google.protobuf.Timestamp ts = 7;
    string conn_name = 10;
}

message Z002 {
    string code = 1;
    string msg = 2;
    string level = 3;
    string invocation_id = 4;
    int32 pid = 5;
    string thread_name = 6;
    google.protobuf.Timestamp ts = 7;
    string exc = 10;
}
